File: src/main/java/ru/ifmo/is/lab1/batchoperations/dto/BatchOperationDto.java
package ru.ifmo.is.lab1.batchoperations.dto;

import lombok.Data;
import ru.ifmo.is.lab1.batchoperations.Status;
import ru.ifmo.is.lab1.users.User;
import java.time.ZonedDateTime;

@Data
public class BatchOperationDto {
  private int id;
  private Status status;
  private Integer addedCount;
  private Integer updatedCount;
  private Integer deletedCount;
  private String errorMessage;
  private User createdBy;
  private ZonedDateTime createdAt;
}

File: frontend/src/interfaces/dto/batchoperations/BatchOperationDto.ts
import User from "../../models/User";

export interface BatchOperationDto {
    id: number;
    status: BatchOperationStatus;
    addedCount: number | null;
    updatedCount: number | null;
    deletedCount: number | null;
    errorMessage: string | null;
    createdBy: User;
    createdAt: string;
}

export enum BatchOperationStatus {
    IN_QUEUE,
    IN_PROGRESS,
    SUCCESS,
    FAILED,
}